app/models/billboard.rb

```diff
@@ -101,6 +103,28 @@ def self.for_display(area:, user_signed_in:, user_id: nil, article: nil, user_ta
     end
   end
 
+  def self.weighted_random_selection(relation)
+    base_query = relation.to_sql
+    random_val = connection.select_value("SELECT RANDOM()").to_f
+
+    record_attrs = connection.select_all(<<-SQL).first
```

Comment:
It seems like this would make more sense as a [`find_by_sql`](https://api.rubyonrails.org/classes/ActiveRecord/Querying.html#method-i-find_by_sql) — unless I'm missing a reason why that wouldn't work?

Reply:
I wrote this in the first way I thought to do it. Just pushed up a refactor in the cleaner ActiveRecord way — along with the Ruby-generated random number. 👍